File: game-server/src/main/java/org/forfun/mmorpg/game/ai/fsm/Transition.java
package org.forfun.mmorpg.game.ai.fsm;

import org.forfun.mmorpg.game.scene.actor.Creature;

public abstract class Transition {

	/** 开始状态 */
	private State from;
	/** 结束状态 */
	private State to;

	public Transition(State from, State to) {
		this.from = from;
		this.to = to;
	}

	/**
	 * 条件判定
	 * 
	 * @param creature
	 * @return
	 */
	public abstract boolean meetCondition(Creature creature);

	public State fromState() {
		return this.from;
	}

	public State toState() {
		return this.to;
	}

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/ai/fsm/Transition.kt
package org.forfun.mmorpg.game.ai.fsm

import org.forfun.mmmorpg.game.scene.actor.Creature


abstract class Transition(private val from: State, private val to: State) {

    /**
     * 条件判定
     * @param creature
     * @return
     */
    abstract fun meetCondition(creature: Creature): Boolean

    fun fromState(): State = this.from

    fun toState(): State = this.to
}
